### PR TITLE
Add Provider and Schema for QuestDB

### PR DESCRIPTION
**Pull request by SuriZhang**
Add Provider and Schema classes for QuestDB.
Generators are only skeletons for now. They will be implemented in the upcoming PRs.

QuestDB is connected via PostgreSQL JDBC with default setting: https://questdb.io/docs/reference/api/postgres/

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -347,6 +347,11 @@
       <artifactId>arangodb-java-driver</artifactId>
       <version>6.9.0</version>
     </dependency>
+    <dependency>
```
Comment (mrigger) [inline pom.xml:350]: Perhaps in this PR, you could only add the questdb dependency. Updating the other dependencies might have some unintended side effects (e.g., CI test failures).

File: src/sqlancer/questdb/QuestDBOptions.java [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+        + " default host: " + QuestDBOptions.DEFAULT_HOST + ")")
+public class QuestDBOptions implements DBMSSpecificOptions<QuestDBOracleFactory> {
+    public static final String DEFAULT_HOST = "localhost";
+    public static final int DEFAULT_PORT = 8812;
+
+    @Parameter(names = "--oracle")
+    public QuestDBOracleFactory oracle = QuestDBOracleFactory.NOREC;
+
+    public enum QuestDBOracleFactory implements OracleFactory<QuestDBGlobalState> {
+        NOREC {
```
Comment (mrigger) [inline src/sqlancer/questdb/QuestDBOptions.java:27]: Perhaps you can keep the enum empty for now? Later on, you can decide if you want to add one or multiple oracles.

**Review by mrigger — COMMENTED** [review]
Thanks a lot for the PR. Could you take a look at the minor comments, and amend the current commit with the changes? The CI tests fail due to some errors that you can reproduce with `mvn verify -DskipTests=True`

File: pom.xml (hunk 1 above)
Comment (mrigger) [reply pom.xml:350]: It seems all the checks pass, so it might be fine to keep the updates.

Comment (SuriZhang) [reply pom.xml:350]: CI test was the only one failed. Anyways I reverted back the version bump.

File: src/sqlancer/questdb/QuestDBOptions.java (hunk 2 above)
Comment (SuriZhang) [reply src/sqlancer/questdb/QuestDBOptions.java]: Yes, fixed this : )

**Pull request merged**